packages/shared/lib/wallet.ts

```diff
@@ -2060,3 +2058,27 @@ export const hasPendingTransactions = (account: WalletAccount): boolean => {
 
     return account?.messages.some((m) => !m.confirmed)
 }
+
+/**
+ * Determines whether an account has any valid pending transactions i.e. transactions that can confirm.
+ *
+ * @method hasValidPendingTransactions
+ *
+ * @param {WalletAccount} account
+ *
+ * @returns {boolean}
+ */
+export const hasValidPendingTransactions = (account: WalletAccount): boolean => {
+    if (!account) return false
+    const pendingMessages = account?.messages.filter((m) => !m.confirmed)
+    const pendingInputs = pendingMessages.flatMap((msg) => {
+        if (msg.payload?.type === 'Transaction') {
+            return msg.payload?.data?.essence?.data?.inputs
+        }
```

Comment:
It should have a default return as well outside `if` because if the payload type is not `Transaction` then it will return `undefined` values.